### ISSUE TITLE
Issue in session disconnection and connecting new session

### ISSUE THREAD

**Issue by KiranKumar36** [LogmeinRescue/iOS-SDK#37]
**Environment**
Xcode 10.2
Swift 4.2
iPhone 7
iOS 12.2

**Description**

Our observations

Issue 1: Technician can't able to view the device camera streaming when starting the camera share session again with a new pin code. 

Issue 2: when starting the screen share session after finishing the camera share session, Sdk throws error "invalidPincodeForSessionType" 

For both of the issues, we think the session is not clearing/ending properly. 
Do we have any option/method to clear session?

**Comment by tibiur** [comment]
thank you for this information, we will investigate these issues

**Comment by rachitasingh1209** [comment]
@tibiur can we please get an update on this issue? This is currently roadblocking us from wrapping up LMI work on iOS on our end.

**Comment by KiranKumar36** [comment]
@tibiur any update on this issue? Please let me know if needed any more information from our side

**Comment by tibiur** [comment]
We are working on a solution, please wait for release 5.1

**Comment by tibiur** [comment]
Issue 1: I could not reproduce it, I just started 5 sessions in a row with our camera example app. Whether I disconnected the session from the technician console or client side, I could always start a new session with PIN code and streaming started without any problem.

Please check the camera example app, what are you doing differently?

**Comment by tibiur** [comment]
@rachitasingh1209 @KiranKumar36

Issue 2: I could not reproduce it.

I used the camera example app with a little modification:

@IBAction func flashlightButtonPressed() {
        // Enable display streaming with audio
        RescueDisplay.sharedInstance.enable()
}
    
@IBAction func pauseButtonPressed() {
        // Enable camera streaming
        RescueCamera.sharedInstance.enable(withAudio: true)
}

So if I tap the flaslight button before starting a session it enables display streaming, if I tap Pause it enables camera streaming. This way I can start different streaming sessions as many times as I want. It is important to enable RescueDisplay.sharedInstance or RescueCamera.sharedInstance before you start the session. Also make sure when you create the session in the technician console you set the proper session type.

**Comment by rachitasingh1209** [comment]
@KiranKumar36 please review the above responses from LMI and reply as soon as you can. Thanks.

**Comment by KiranKumar36** [comment]
@tibiur  we cross verified the integration and everything looks as per doc and example app.
We could reproduce both the issues often in our app. 

For you to reproduce, we have combined both the example apps(camera share and screen share) into one app and could replicate both the issues.
Please find the example project [here](https://www.dropbox.com/s/51m3g4d8ignbkb0/LogMeIn-iOS-SDK-master.zip?dl=1)

**Steps for issue1**
1. Connect to camera share with pin code 1
2. Disconnect camera share
3. Broadcast screen share with pin code 2
4. Disconnect screen share
5. Connect again to camera share with pincode 3
6. Session says it connected but techinician can't able to view session.

*Steps for issue2**
1. Broadcast screen share with pin code 1
2. Disconnect screen share
3. Connect again to camera share with pincode 2
4. Session stays in connecting state

**Comment by tibiur** [comment]
I think you set RescueSession.sharedInstance.connectToSessionAutomatically to false when using broadcast session and never set to true before you try to connect with camera streaming. That's why it's stuck in connected state. 

I just added RescueSession.sharedInstance.connectToSessionAutomatically = true to CameraViewController's viewDidLoad() and I was able to connect with camera and broadcast multiple times without any problem.
